### DISCUSSION TITLE
Architectures of all examples

### DISCUSSION THREAD

**Discussion by dang-trinh156** [lululxvi/deepxde#1271] [Q&A] [answered]
Hi Mr.LuLu and DeepXDE users, 
I have gone through several examples of DeepXDE, I only see architectures that have already specified. How can you know how many layers, how many neurons per layer to specify in those examples?
Thank you for your answers!

**Comment by pescap** [comment]
Hi,
For example, in:
https://github.com/lululxvi/deepxde/blob/bc9cf248e747184f18c0cf15851ad24bb2efc143/examples/pinn_forward/Poisson_Dirichlet_1d.py#L35

the fully connected neural network has depth 4 (i.e., 3 hidden layers) and width 50 (also refer to [this tutorial](https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/poisson.1d.dirichlet.html)).

`[1] + [50] * 3 + [1]` means:
- Input layer of size 1
- 3 hidden layers of 50 neurons
- Output layer of size 1.

**Reply by dang-trinh156** [reply]
Hi,
Thank you for that. But how can we know how many layers and neurons?, I mean the hyperparameters for these examples

**Reply by pescap** [reply, marked as answer]
As far as I know, the choice is generally coming from empirical observations. You can also perform hyper-parameters tuning, e.g following the structure in:
https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/helmholtz.2d.dirichlet.hpo.html
